metals/src/main/scala/scala/meta/internal/metals/ReferenceProvider.scala

```diff
@@ -196,7 +196,7 @@ final class ReferenceProvider(
       includeSynthetics: Synthetic => Boolean
   ): Seq[Location] = {
     val isSymbol = alternatives + occ.symbol
-    if (occ.symbol.isLocal) {
+    if (occ.symbol.isLocal || source.isDependencySource(workspace)) {
```

Comment:
This will make sure that when searching for references in dependency sources it will search the current file despite those sources not being indexed.